rviz_default_plugins/CMakeLists.txt

```diff
@@ -341,10 +327,30 @@ if(BUILD_TESTING)
     visualization_msgs
   )
 
+  # None of the tests are Qt files, so we can save a bit of time by not letting
+  # AUTOMOC run on every one of them and generate an empty file.
+  set(CMAKE_AUTOMOC OFF)
+
+  # display_test_fixture.cpp is a piece of code that is used by every test.
+  # Instead of recompiling it for every test, just compile it once as a library
+  # and reuse it.  Note that we need to call ament_find_gmock() by hand here to
+  # find GMOCK_INCLUDE_DIRS so we can compile this file.
```

Comment:
Perhaps we need something like this but for libraries:

https://github.com/ament/ament_cmake/blob/master/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake

In the meantime you should double check all the things it does. 

It would be better to not have gtest stuff in a separate library. Maybe the fixture could be changed so gtest is not used in it. There were some gotchas about compiling gtest into libraries if I remember correctly. I don’t know if those apply here.

Reply:
> Perhaps we need something like this but for libraries:
> 
> https://github.com/ament/ament_cmake/blob/master/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake

Yeah, there is already https://github.com/ament/ament_cmake/blob/master/ament_cmake_gmock/cmake/ament_add_gmock.cmake , but it only does executables.  I agree that the addition of a `ament_add_gmock_library()` would be a good thing to have; I'll open an issue for it.

> In the meantime you should double check all the things it does.

Will do.

> It would be better to not have gtest stuff in a separate library. Maybe the fixture could be changed so gtest is not used in it. There were some gotchas about compiling gtest into libraries if I remember correctly. I don’t know if those apply here.

That's a good idea.  Looking at it briefly, it doesn't look trivial to do.  I'll open an issue to follow-up with that as well.